task_four.py

```diff
@@ -0,0 +1,117 @@
+# Задание
+#
+# Написать программу, которая скачивает изображения с заданных URL-адресов и сохраняет их на диск.
+# Каждое изображение должно сохраняться в отдельном файле, название которого соответствует
+# названию изображения в URL-адресе.
+# Например, URL-адрес: https://example/images/image1.jpg -> файл на диске: image1.jpg
+# — Программа должна использовать многопоточный, многопроцессорный и асинхронный подходы.
+# — Программа должна иметь возможность задавать список URL-адресов через аргументы командной строки.
+# — Программа должна выводить в консоль информацию о времени скачивания каждого
+# изображения и общем времени выполнения программы.
+
+import time
+import os
+import requests
+import threading
+import multiprocessing
+import asyncio
+import aiohttp
+from aiohttp import ClientSession
+
+urls = ['https://w.forfun.com/fetch/7d/7db7d55995bd8690e98b42b2d57aa496.jpeg',
+        'https://mykaleidoscope.ru/x/uploads/posts/2022-10/1666206272_76-mykaleidoscope-ru-p-kartinka-na-zastavku-oboi-79.jpg',
+        'https://get.wallhere.com/photo/Paris-beautiful-france-Eiffel-Tower-city-France-1091947.jpg',
+        'https://proprikol.ru/wp-content/uploads/2019/10/delfiny-krasivye-kartinki-27.jpg',
+        'https://mykaleidoscope.ru/x/uploads/posts/2022-10/1666206216_27-mykaleidoscope-ru-p-kartinka-na-zastavku-oboi-27.jpg'
+        ]
+
+
+def download(url, filename):
+    responce = requests.get(url).content
+    with open(filename, "wb", ) as f:
+        f.write(responce)
+
+
+def synch_method():
+    folder = 'data_task_four'
+    start_time = time.time()
+    if not os.path.exists(folder):
+        os.mkdir(folder)
+    for url in urls:
+        name = os.path.join(folder, url.split('/')[-1])
```

Comment:
Чтобы не длить все, можно использовать метод `rsplit`
```suggestion
        name = os.path.join(folder, url.rsplit('/', 1)[1])
```
или так
```suggestion
        name = os.path.join(folder, os.path.basename(url))
```